src/eko/evolution_operator/grid.py

```diff
@@ -118,7 +127,7 @@ def get_threshold_operators(self, path: List[Segment]) -> List[Operator]:
         is_downward = is_downward_path(path)
         shift = flavor_shift(is_downward)
         for seg in path[:-1]:
-            new_op_key = astuple(seg)
+            new_op_key = seg
```

Comment:
Great that you removed the `astuple()` call, but do we need the `new_op_key` variable any longer?